### PR TITLE
Fix for early exit out of ParseCertRelative

### PR DESCRIPTION
**Pull request by lealem47**
# Description

Reproducer is posted as an internal note in zd16775

Fixes zd#16775

# Testing

How did you test?

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: wolfcrypt/src/asn.c [hunk 1]
```diff
@@ -23198,7 +23198,7 @@ int ParseCertRelative(DecodedCert* cert, int type, int verify, void* cm)
 #endif
             {
                 WOLFSSL_ERROR_VERBOSE(ASN_NO_SIGNER_E);
-                return ASN_NO_SIGNER_E;
+                ret = ASN_NO_SIGNER_E;
```
Comment (dgarske) [inline wolfcrypt/src/asn.c:23201]: Should the `return ASN_SELF_SIGNED_E;` above also be changed?

Comment (lealem47) [reply wolfcrypt/src/asn.c:23201]: It definitely should. Thanks for catching that

Comment (dgarske) [reply wolfcrypt/src/asn.c:23201]: There are other `return ret` cases above too. Please review if those need changed also.

Comment (lealem47) [reply wolfcrypt/src/asn.c:23201]: I can change them for consistencies sake but I believe the `else` case returning `ASN_SELF_SIGNED_E` or `ASN_NO_SIGNER_E` were the only points that yielded undesired behavior

Comment (dgarske) [reply wolfcrypt/src/asn.c:23201]: Great, then only change these two. Can you document what the undesirable behavior is here?

Comment (lealem47) [reply wolfcrypt/src/asn.c:23201]: Originally I thought this was a matter of error prioritization but this PR is just wrong. Customer was using verify callbacks with the cert manager to override date errors when loading a CA but I now realize that adding a CA and simply overwriting the return value regardless of the result is invalid. To get what they want to do to work, some logic will on the verify  [here](https://github.com/wolfSSL/wolfssl/blob/210fff55695141ca8797b5644a8215aab49f2e5a/src/ssl.c#L5740) will have to be added based on if `cm->verifyCallback` is set. I've instead recommended the use of `wolfSSL_CertManagerLoadCABuffer_ex()`

**Comment by lealem47** [comment]
Closing in favor of `wolfSSL_CertManagerLoadCABuffer_ex()` approach

**Pull request closed without merging**